### ISSUE TITLE
Add ability to drop in scripting from other JVM languages

### ISSUE THREAD

**Issue by cmungall** [ontodev/robot#149]
Robot provides a menu of verbs of the most common operations. It's also useful to have custom programmatic transformations. Some can be done with the SPARQL command, but not all.

It would be great to drop in jvm commands with no boilerplate, with the appropriate owlapi wrapper already imported.

E.g. https://github.com/phenoscape/scowl#add-some-axioms-and-programmatically-generated-gcis-to-an-ontology

```
val HasPart = ObjectProperty("http://example.org/has_part")
val DevelopsFrom = ObjectProperty("http://example.org/develops_from")
val gcis = for {
  term <- ontology.getClassesInSignature(true)
} yield {
  (not(HasPart some term)) SubClassOf (not(HasPart some (DevelopsFrom some term)))
}
manager.addAxioms(ontology, gcis)
```

we can imagine similar for clojure, groovy, etc.

maybe this doesn't make much sense and it would be better to do this directly from a program written in the high level language of choice. but just like our sparql wrapper, it's convenient having everything available in one executable, no compilation, no boilerplate.

cc @balhoff

**Comment by jamesaoverton** [comment]
I like the idea in the abstract. As soon as I start thinking of details, every question leads to ten more.

I use ROBOT in Makefiles for many projects. I would prefer to skip Make and just have an ontology build script that I can run without any fuss. There's a million ways to do it, and I've tried a bunch of experiments, but I don't have a solution that I like yet.

**Comment by cmungall** [comment]
Yep, I personally like makefiles but they're harder for a lot of the robot community to use.

The feature I had in mind was less about replacing control flow and dropping in additional domain-specific commands. But the two are related. If we did switch the flow to within ROBOT we'd want the ability to do scripting extensions.

**Comment by cmungall** [comment]
I guess the main idea here was not to replace makefiles, but as drop ins for individual steps within a pipeline.

For example, in the GO pipeline we want to run a check that no `consider` annotation points to a deprecated class. This is awkward as the consider annotations are CURIEs as string literals. I believe it should be possible as SPARQL like this:

```
SELECT ?x ?r ?clsIdentifier WHERE {
       ?cls a owl:Class  ;
	    owl:deprecated "true"^^xsd:boolean .
       ?x ?r ?clsIdentifier
       FILTER (str(?clsIdentifier) =
               replace(str(?cls),"http://purl.obolibrary.org/obo/GO_","GO:"))
}
```

but this doesn't seem to terminate.. it could probably be done as a couple of lines of scala, cc @balhoff
